src/Sylius/Behat/Context/Api/Admin/ManagingShippingCategoriesContext.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Component\Shipping\Model\ShippingCategoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ManagingShippingCategoriesContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $client;
+
+    public function __construct(ApiClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @Given I want to create a new shipping category
+     */
+    public function iWantToCreateANewShippingCategory(): void
+    {
+        $this->client->buildCreateRequest('shipping_categories');
+    }
+
+    /**
+     * @Given I want to modify a shipping category :shippingCategory
+     */
+    public function iWantToModifyAShippingCategory(ShippingCategoryInterface $shippingCategory): void
+    {
+        $this->client->buildUpdateRequest('shipping_categories', $shippingCategory->getCode());
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt(): void
+    {
+        $this->client->create();
+    }
+
+    /**
+     * @When I delete shipping category :shippingCategory
+     */
+    public function iDeleteShippingCategory(ShippingCategoryInterface $shippingCategory): void
+    {
+        $this->client->delete('shipping_categories', $shippingCategory->getCode());
+    }
+
+    /**
+     * @When I browse shipping categories
+     */
+    public function iBrowseShippingCategories(): void
+    {
+        $this->client->index('shipping_categories');
+    }
+
+    /**
+     * @When I do not specify its code
+     * @When I specify its code as :code
+     */
+    public function iSpecifyItsCodeAs(?string $code = null): void
+    {
+        if ($code !== null) {
+            $this->client->addRequestData('code', $code);
+        }
+    }
+
+    /**
+     * @When I name it :name
+     * @When I do not specify its name
+     * @When I rename it to :name
+     */
+    public function iNameIt(?string $name = null): void
+    {
+        if ($name !== null) {
```

Comment:
I'm wondering if we shouldn't allow for assigning `null` to requested keys

Reply:
In first iteration I added `string $name = ''`, but after https://github.com/Sylius/Sylius/pull/11141#discussion_r383919292 suggestion I changed it